airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -304,7 +323,7 @@ def check_for_key(self, key, bucket_name=None):
 
     @provide_bucket_name
     @unify_bucket_name_and_key
-    def get_key(self, key, bucket_name=None):
+    def get_key(self, key: str, bucket_name: Optional[str] = None) -> S3Transfer:
```

Comment:
This seems incorrect to have annotated with return type S3Transfer though actual return type is boto3.s3.Object. This is giving pycharm fits due to the mismatched interface. Hopefully no impact at runtime.

Reply:
Interesting. Could you share the output you are seeing from pycharm? I cannot figure out where the `boto3.s3.Object` is defined. I have been through the `boto3`, `botocore` and `airflow` code as well as [some docs]() but I have not been able to.

I think we need the [`boto3-type-annotations`](https://pypi.org/project/boto3-type-annotations/) to solve this issue. [Check out this Stack Overflow question also](https://stackoverflow.com/questions/52087307/adding-type-hinting-to-functions-that-return-boto3-objects).

This is something `@mik-laj` once brought up in https://github.com/apache/airflow/issues/11297 it looks like. I don't have time to tackle it right now, but I could in the near future.

Reply:
Just dropped a comment over in #11297 as well: https://github.com/apache/airflow/issues/11297#issuecomment-864488490

Reply:
Thanks, Cooper. Saying it's giving pycharm fits might have been hyperbole by me. It's showing an error message due to the confusion over types:

> Unresolved attribute reference 'download_fileobj' for class 'S3Transfer'

When I step into the get_key function it's showing rtype as Object:

>  :rtype: boto3.s3.Object

Reply:
https://pypi.org/project/boto3-stubs/ might be a helpful project here.